### PR TITLE
Implement `bin` and `hex` builtin functions

### LINKED ISSUE #174: Implement the builtin `hex` function
This is a good first issue to get your hands dirty in the rustpython sourcecode.

Hit `cargo run`, and try the `hex` function. It does not exist yet, so please add it to `vm/src/builtins.rs`.

### PR DESCRIPTION
**Pull request by cauebs**
Closes #174 

- [X] Implement
- [X] Add tests
- [ ] Document?

### CODE REVIEW

**Comment by windelbouwman** [comment]
@cauebs thank you for adding this code to rustpython! I noticed still tests failed. I think you can check if the value of the integer is negative, and handle it accordingly.

File: vm/src/builtins.rs [hunk 1]
```diff
@@ -74,7 +74,21 @@ fn builtin_any(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
 }
 
 // builtin_ascii
-// builtin_bin
+
+fn builtin_bin(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(number, Some(vm.ctx.int_type()))]);
+
+    let n = match number.borrow().kind {
```
Comment (windelbouwman) [inline vm/src/builtins.rs]: There is a helper function you could use here: `let n = objint::getvalue(number);`.

**Comment by windelbouwman** [comment]
One other note: if you have closed this issue, could you also create a new good starter issue to help other people get started? Maybe this way we could create a sort of chained issue for new contributors!

**Comment by cauebs** [comment]
@windelbouwman I love that idea. There are some other builtins that might be simple and I'll make sure to open an issue pointing to one ;)

File: vm/src/builtins.rs (hunk 1 above)
Comment (cauebs) [reply vm/src/builtins.rs]: Done. Thanks! :)

**Comment by cauebs** [comment]
Ok, I have no idea of what's going wrong :(

File: tests/snippets/builtin_bin.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+assert bin(0) == '0b0'
+assert bin(1) == '0b1'
+assert bin(-1) == '-0b1'
+assert bin(2**65) == '0b1' + '0' * 65
```
Comment (windelbouwman) [inline tests/snippets/builtin_bin.py]: At the moment we use 32 bit integers. There is an open point to change this to multiprecision arithmatic (for example https://gmplib.org/). For now, mark this test case as TODO by commenting and placing a TODO comment for it. We can only use 2**31 and less for now.

**Pull request merged**